content/using/id/overview.md

```diff
@@ -7,65 +7,189 @@ template = "doc.html"
+
+Existing PKIs include PGP-style ["web of
+trust"](https://en.wikipedia.org/wiki/Web_of_trust), [SSL
+certificates](https://en.wikipedia.org/wiki/Certificate_authority),
+[ZeroTier](https://www.zerotier.com/lf-announcement/),
+[Keybase](https://keybase.io/), [OpenID](https://openid.net/what-is-openid/),
+[Mozilla
+Persona](https://developer.mozilla.org/en-US/docs/Archive/Mozilla/Persona), and
+[Login with Google](https://developers.google.com/identity). These take unique
+approaches to the problem and have achieved some degree of success, but none
+provide globally consistent, permanent, and completely self-owned identities.
+Exhaustive exploration and categorization is unfortunately out of scope for this
+post, so we'll just describe Urbit's approach to achieving these properties in
+our PKI.
+
+In Urbit, a "name" is often called a "[ship](/docs/glossary/ship)" or an "address" because we use the
+metadata in the PKI to make names routable. The total data is two 256-bit
+asymmetric keys, a cryptographic suite number (to allow changing crypto
+algorithms), the revision number of the key, and the name of a ship that will
+route for it. This sums to less than 128 bytes of data.
+
+Each PKI trades off various properties. We chose a tripartite system so that
+appropriate choices can be made for different use cases. Here, we explore the
+various properties we chose by following a series of binary choices -- the idea
+maze.
+
+There are two sorts of PKI: either you can change your keys or not. If you
+cannot, then your name is impermanent, for no one can keep a set of keys secure
+forever. Even if your opsec never fails, eventually your crypto algorithms may
+be compromised.
+
+However, if you sacrifice the ability to change your keys, you can have a very
+nice property: self-attesting keys. If your name is a hash of your public key
+information, then no other source of information is required to verify you are
+who you say you are. This is a useful property since it requires no coordination
+at all. Many things are impermanent, especially during development. It also
+provides a way to try out the network without obtaining a permanent address.
+
+This is our first stop in the idea maze. We call this sort of name a "[comet](/docs/glossary/comet)". A
+comet's name is a 128-bit human-readable hash of its public keys.
+
+However, Urbit is yours and it's forever. You shouldn't have to change your name
+every time you change your keys. So, we go back and take the other choice as well: you
+must be able to change your keys.
+
+To change keys, you must sign a message with the old keys revoking them and
+supplying the new ones[^0]. The question is what happens if the old keys also
+sign a second set of new keys. This could happen if an attacker obtained your
+old keys after the fact. This is important because one of the reasons to be able
+to change your keys is to invalidate the old ones so that they have no power.
```

Comment:
```suggestion
<a name="0"></a>
To change keys, you must sign a message with the old keys revoking them and
supplying the new ones<sup><a href="#f0">[0]</a></sup>. The question
is what happens if the old keys also sign a second set of new keys. This could
happen if an attacker obtained your old keys after the fact. This is important
because one of the reasons to be able to change your keys is to invalidate the
old ones so that they have no power.
```

Reply:
add footnote links